### PR TITLE
QA-63 | Detox setup UI E2E tests

### PR DESCRIPTION
**Pull request by eugenezhernakov**
* Add and setup Detox test framework
* Some Android-specific changes as per [docs](https://wix.github.io/Detox/docs/introduction/project-setup)
* Add test IDs to various components for easy access during testing
* Working CI setup for running on iOS

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -101,9 +101,9 @@ jobs:
 #      - restore_cache:
 #          keys:
 #            - judokit-reactnative-sample-ios-cocoapods-{{ checksum "sample-app/ios/Podfile" }}
-      - restore_cache:
-          keys:
-            - judokit-reactnative-sample-ios-ruby-gems-{{ checksum "sample-app/Gemfile.lock" }}
+      # - restore_cache:
+      #     keys:
+      #       - judokit-reactnative-sample-ios-ruby-gems-{{ checksum "sample-app/Gemfile.lock" }}
```
Comment (stefan-tudor) [inline .circleci/config.yml:106]: Is this necessary?

File: sample-app/e2e/functional.test.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  //   // Will fail until CSC toggle is implemented
+  //   await assertResultsScreen('1', '1')
+  // })
+
+  // it('should successfully complete a 3DS2 pre-auth transaction via payment methods', async () => {
+  //   await element(by.text(Selectors.PREAUTH_METHODS)).tap()
+  //   await addCardPaymentMethod()
+  //   // Will fail until CSC toggle is implemented
+  //   await assertResultsScreen('1', '1')
+  // })
```
Comment (stefan-tudor) [inline sample-app/e2e/functional.test.ts:104]: `Will fail until CSC toggle is implemented` - it is implemented, 
IMG_6302

File: .circleci/config.yml (hunk 1 above)
Comment (eugenezhernakov) [reply .circleci/config.yml:106]: Honestly I am not sure, do we want it to pull down gems/pods/deps every time?

File: sample-app/e2e/functional.test.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      TestData.SECURITY_CODE,
+    )
+    await element(by.id(Selectors.TOKEN_SCROLL_VIEW)).scrollTo('bottom')
+    await element(by.id(Selectors.PAY_WITH_TOKEN)).tap()
+    await complete3DS2()
+    try {
+      await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+      await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+    } catch (error) {}
+    await assertResultsScreen('1', '1')
```
Comment (stefan-tudor) [inline sample-app/e2e/functional.test.ts:43]: ` assertResultsScreen('1', '1')` - these are magic numbers, and it's unclear what is what,
I would recommend using well-named enums or constants for this purpose and maybe having something like this is more clear:


```
export interface Props {
    type: string,
    result: string
}

export async function assertResultsScreen(props: Props) => {}


then you can invoke it like so:

await assertResultsScreen({ type: "1", result: "1" })
```

File: sample-app/e2e/functional.test.ts (hunk 2 above)
Comment (eugenezhernakov) [reply sample-app/e2e/functional.test.ts:104]: I think we have a bug then, as I don't see the CSC sheet to enter the CV2 come up when trying a payment via payment methods, can you try on your end?

File: sample-app/e2e/functional.test.ts (hunk 3 above)
Comment (eugenezhernakov) [reply sample-app/e2e/functional.test.ts:43]: Superb, thank you for this! I'll implement it this way

Comment (eugenezhernakov) [reply sample-app/e2e/functional.test.ts:43]: Superb, thank you for this! I'll implement it this way

**Review by julia-feltham — APPROVED** [review]

File: sample-app/e2e/functional.test.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  //   await dissmissKeyboardOnTokenScreen()
+  //   await element(by.id(Selectors.TOKEN_SCROLL_VIEW)).scrollTo('bottom')
+  //   await element(by.id(Selectors.PREAUTH_WITH_TOKEN)).tap()
+  //   await complete3DS2()
+  //   try {
+  //     await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+  //     await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+  //   } catch (error) {}
+  //   await assertResultsScreen({type: '2', result: '1'})
+  // })
```
Comment (RafalOzogJudopay) [inline sample-app/e2e/functional.test.ts:65]: What about these commented out code in this class?

**Pull request closed without merging**